sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthorizationCodeCredentialBuilder.java

```diff
@@ -47,6 +50,40 @@ public AuthorizationCodeCredentialBuilder redirectUrl(String redirectUrl) {
         return this;
     }
 
+    /**
+     * Sets the client secret for the authentication.
+     * @param clientSecret the secret value of the AAD application.
+     * @return the AuthorizationCodeCredentialBuilder itself
+     */
+    public AuthorizationCodeCredentialBuilder clientSecret(String clientSecret) {
```

Comment:
It's funny since in the beginning .NET and Python supporting providing a client secret and Java didn't. Now Java supports providing a client secret or a certificate :)